test/archethic_web/aeweb/domain_test.exs

```diff
@@ -0,0 +1,233 @@
+      4J6i0XG8stulp5XpXIKBY7ltQZpTfvCInFm1ztaVDpAaXj0CZ4p9GL+u2heoUm8i
+      ZC0qo6pSt01bfbaOpLKt8ERuauXTg6iiV4t5XLjTH5AZrsoB2zKrb7LvnCJe94fC
+      1YSlraaD1wuEoUu56JetbmUsyE3uYKfNlSnwWRJ9C5zKz75wQ8VRqtjDzgPowZ0L
+      czxOPvP/AgMBAAECgf9L1kDmNDlBN4k7B+BbYZrYs7lUDlIqjALr0ZLjTJdg9tL+
+      exHSwEtWi9rpXdceEx0s4U3v60AzteUFfiNE2DoS1RO0l1AiGcfXb51Pfe6JnNRi
+      PO1p5699rUvFVeE15+lUViPVWU+uma3y+s5IwcIQV3redqyXS6M7izyKMVU7mBTK
+      e63pDSu2cxODfE9p3M/Mvk1uaf+OQ4qBDqsh45q6QDZWMuVw+iitt+vMub8a2EzS
+      HryRSGcfg9zgvcOcOKjp1u/UjAsO2tCfXMZLQoWEG8RmFOnPqVlh2ZXfmCvtCP1W
+      K2fUuOmc0ddFBhO3vk5gqO5Jsh+CVj6sjTZlaRkCgYEA/cQBbhoYO4D+diDr3D5N
+      01Z+kZ4HmgrW6gahef+MNy9YWXooghlcKe7IYXlBqsMI/IiyK9zbMAcOK61zi/Le
+      EezoXbjdk+6VvuniPJEWX/JquJoqiTmDwzP0lQEM/aKdOG624zYfObjJ6aBaf/xk
+      ssdHRhOfGp9pFSaHA0KO3PcCgYEA9lu+b/n2v0JeEG7IR1RzJ1Ud4DhgZyvvIggf
+      hxX5tMZ0E8ywOnJ4g2o3qowI4XJmDCudI29cmTNgVKqwUduDgOoAkdlpoWvhY3yb
+      DzmC8Fim+/MAOGGPyRAuPovBm+dGH0fQnatS0tuo5MpKRB2BnJ3RrJ4Qm1f3G/we
+      gA6DBzkCgYB+9VUR1JRTENI+H3JhGfqtxRRFnh6HfuzO4Mpg0u0/nrxA59DkZfOq
+      NwChY5zq5fDVBz68mx4+BQmd6IVqevOHXFNUsGyK2k6o2TKKwrvC/PFPsjGdvdyi
+      CJhRA9mP+49U8G8ndahhpIXAEK22Ynuuxexurtpm42IbZs8dXmtDOQKBgQC9fbnQ
+      VXsWh8zkZOHGA84DHfQ56AM2uFNaYNcnR57nDpJwPEv82Nmbc1LX6phWGHEnwVA/
+      1kNqT1s0JIo0nFzdBqBjjtAx6lHV/R0jq7/scLQYLUQpGdnH9JstXsAP0+da3hk3
+      fXTaXTzepj5TgEKWncmONZJeel3G97jaFM9x+QKBgQCWW3sPWKh/9Z5LSn48irDF
+      88dy1WqPPZXllilNNvrTA7bMfXQqN14doFTMcWaAUoEDl7H/sWSuBFFQddYKsLSR
+      SC9ttOp9kKUYgCxGmbE3Fwj8LsuyddhGisZ0edC2rJvVvCQMCIgUS9VvSnpzpiay
+      rgoVgtbapk/vMon8gnjqMw==
+      -----END PRIVATE KEY-----
+
+      """
+
+      https_conf =
+        :archethic
+        |> Application.get_env(ArchethicWeb.Endpoint)
+        |> Keyword.fetch!(:https)
+
+      keyfile = Keyword.fetch!(https_conf, :keyfile)
+      certfile = Keyword.fetch!(https_conf, :certfile)
+
+      unlisted_domain_key_pem = File.read!(Application.app_dir(:archethic, keyfile))
+
+      unlisted_domain_cert_pem = File.read!(Application.app_dir(:archethic, certfile))
+
+      %{
+        fake_cert_pem: fake_cert_pem,
+        genesis_address: genesis_address,
+        fake_key_pem: fake_key_pem,
+        unlisted_domain_cert_pem: unlisted_domain_cert_pem,
+        unlisted_domain_key_pem: unlisted_domain_key_pem
+      }
+    end
+
+    test "should return the correct key and cert for a domain", %{
+      fake_cert_pem: fake_cert_pem,
+      genesis_address: genesis_address,
+      fake_key_pem: fake_key_pem
+    } do
+      setup_transaction(fake_key_pem, fake_cert_pem, genesis_address)
+
+      result = Domain.sni("example.com")
+
+      expected_key = read_pem(fake_key_pem) |> hd()
+      expected_cert = read_pem(fake_cert_pem) |> hd() |> elem(1)
+      assert [key: expected_key, cert: expected_cert] == result
+
+      result = Domain.sni("blog.example.com")
+
+      assert [key: expected_key, cert: expected_cert] == result
```

Comment:
Here you should add another test to ensure the wildcard (*.example.com) is working well